src/hip/hip_build_utils.cpp

```diff
@@ -39,6 +39,48 @@
 MIOPEN_DECLARE_ENV_VAR_BOOL(MIOPEN_DEBUG_HIP_VERBOSE)
 MIOPEN_DECLARE_ENV_VAR_BOOL(MIOPEN_DEBUG_HIP_DUMP)
 
+#if MIOPEN_OFFLINE_COMPILER_PATHS_V2
+
+// List to hold Relative Directory Path
+// for each Compiler Flags from ROCM Install Path
+// The definition of this variable expected to be available in C/C++
+// code of the Target Library.
+static const char* MIOPEN_CMP_FLAG_REL_PATHS[MIOPEN_CMP_MAX_Flag] = {
+    "llvm/bin/clang", "bin/clang-ocl", "llvm/bin/clang++", "llvm/bin/clang-offload-bundler"};
+
+// Global variable List to hold the Compiler Flag Paths.
+char MIOPEN_CMP_FLAGS[MIOPEN_CMP_MAX_Flag][PATH_MAX_LEN] = {0};
+
+// Function to generate the MIOPEN Compiler Flag Value using
+// ROCm Base Install Path fetched using getROCmInstallPath()
+// This approach depends on the getROCmInstallPath() provided by rocm-core
+// This flag Paths are expected to be deprecated/modified in upcoming MAJOR Releases.
+char* generateFlagValue(MIOPEN_CMP_Flags_t x)
+{
+    char* flagPath   = (char*)0;
+    unsigned int len = 0;
+    if(MIOPEN_CMP_MAX_Flag <= x)
+    {
+        return (char*)0;
+    }
+    if(0 == MIOPEN_CMP_FLAGS[x][0])
+    {
+        PathErrors_t ret = getROCmInstallPath(&flagPath, &len);
+        if(PathSuccess == ret)
+        {
+            if(PATH_MAX_LEN > (len + strlen(MIOPEN_CMP_FLAG_REL_PATHS[x])))
+            {
+                sprintf(
+                    (char*)&(MIOPEN_CMP_FLAGS[x]), "%s%s", flagPath, MIOPEN_CMP_FLAG_REL_PATHS[x]);
+            }
+            free(flagPath);
+        }
+    }
+    return (char*)&(MIOPEN_CMP_FLAGS[x]);
+}
```

Comment:
Please use `fs::path` as a return type. Returning `char*` and not even the `const char*`, you are opening a vast vulnerability and potential easy access for buffer overflow attacks.

Reply:
Also, please don't use C casting. Use C++ casting instead - `static_cast<>(..)` or `reinterpret_cast<>(..)`. Ideally, do not use casting at all.

Reply:
Thanks for the comments. These lines of code changed in latest version